components/merkledb/src/proof_list_index/mod.rs

```diff
@@ -560,9 +569,24 @@ where
     where
         I: IntoIterator<Item = V>,
     {
+        let first_index = self.len();
+        let mut last_index = first_index;
```

Comment:
I think name like `new_list_len` (and `old_list_len` in the line above) will be more clear.